web/src/app/services/data-store/data-store.service.ts

```diff
@@ -311,20 +311,35 @@ export class DataStoreService {
       .pipe(map(data => FirebaseDataConverter.toUser(data as DocumentData)));
   }
 
-  lois$({id}: Survey): Observable<List<LocationOfInterest>> {
-    return this.db
-      .collection(`${SURVEYS_COLLECTION_NAME}/${id}/lois`)
-      .valueChanges({idField: 'id'})
-      .pipe(
-        map(array =>
-          List(
-            array
-              .map(obj => LoiDataConverter.toLocationOfInterest(obj.id, obj))
-              .filter(DataStoreService.filterAndLogError<LocationOfInterest>)
-              .map(loi => loi as LocationOfInterest)
-          )
-        )
-      );
+  lois$({id}: Survey, {email}: User): Observable<List<LocationOfInterest>> {
+    const predefinedLois = this.db.collection(
+      `${SURVEYS_COLLECTION_NAME}/${id}/lois`,
+      ref => ref.where('predefined', '==', true)
+    );
+
+    const userLois = this.db.collection(
+      `${SURVEYS_COLLECTION_NAME}/${id}/lois`,
+      ref =>
+        ref
+          .where('predefined', '==', false)
+          .where('created.user.email', '==', email)
+    );
+
+    return combineLatest([
+      predefinedLois.valueChanges({idField: 'id'}),
+      userLois.valueChanges({idField: 'id'}),
+    ]).pipe(
+      map(lois => {
```

Comment:
Does array arg destructuring work the way I think it does? If so you can do `map([predefinedLois, userLois] => ...)`